jumanji/environments/logic/graph_coloring/env.py

```diff
@@ -0,0 +1,310 @@
+            representing the adjacency matrix of the graph.
+        - colors: jax array (int32) of shape (num_nodes,),
+            representing the current color assignments for the vertices.
+        - action_mask: jax array (bool) of shape (num_colors,),
+            indicating which actions are valid in the current state of the environment.
+        - current_node_index: integer representing the current node being colored.
+
+    - action: int, the color to be assigned to the current node (0 to num_nodes - 1)
+
+     - reward: float, a sparse reward is provided at the end of the episode.
+        Equals the negative of the number of unique colors used to color all vertices in the graph.
+        If an invalid action is taken, the reward is the negative of the total number of colors.
+
+    - episode termination:
+        - if all nodes have been assigned a color or if an invalid action is taken.
+
+    - state: `State`
+        - adj_matrix: jax array (bool) of shape (num_nodes, num_nodes),
+            representing the adjacency matrix of the graph.
+        - colors: jax array (int32) of shape (num_nodes,),
+            color assigned to each node, -1 if not assigned.
+        - current_node_index: jax array (int) with shape (),
+            index of the current node.
+        - action_mask: jax array (bool) of shape (num_colors,),
+            indicating which actions are valid in the current state of the environment.
+        - key: jax array (uint32) of shape (2,),
+            random key used to generate random numbers at each step and for auto-reset.
+
+    ```python
+    from jumanji.environments import GraphColoring
+    env = GraphColoring()
+    key = jax.random.key(0)
+    state, timestep = jax.jit(env.reset)(key)
+    env.render(state)
+    action = env.action_spec().generate_value()
+    state, timestep = jax.jit(env.step)(state, action)
+    env.render(state)
+    ```
+    """
+
+    def __init__(
+        self,
+        generator: Optional[Generator] = None,
+        viewer: Optional[Viewer[State]] = None,
+    ):
+        """Instantiate a `GraphColoring` environment.
+
+        Args:
+            generator: callable to instantiate environment instances.
+                Defaults to `RandomGenerator`.
+                - The default value of `num_nodes` is 100.
+                - The default value of `percent_connected` is 0.8.
+            viewer: callable to instantiate environment viewer.
+                Defaults to `GraphColoringViewer`.
+        """
+        self.generator = generator or RandomGenerator(
+            num_nodes=100, percent_connected=0.8
+        )
+        num_nodes = self.generator.num_nodes
+
+        # Create viewer used for rendering
+        self._env_viewer = viewer or GraphColoringViewer(
+            num_nodes=num_nodes, name="GraphColoring"
+        )
+
+    def __repr__(self) -> str:
+        """Returns: str: the string representation of the environment."""
```

Comment:
Maybe we don't need a doc string for this? I don't recall us doing this in other envs.